assets/js/interactivity/vdom.js

```diff
@@ -60,11 +78,17 @@ export function toVdom( root ) {
 					ignore = true;
 				} else if ( n === islandAttr ) {
 					island = true;
+					try {
+						const val = JSON.parse( attributes[ i ].value );
+						namespace = val?.namespace ?? null;
+					} catch ( e ) {}
 				} else {
 					hasDirectives = true;
 					let val = attributes[ i ].value;
 					try {
-						val = JSON.parse( val );
+						const [ json, ns ] = nsPathParser( val );
+						val = JSON.parse( json );
+						if ( ns && val ) val[ $$namespace ] = ns;
```

Comment:
This is kind of a hack. Directives that hold a string value (a.k.a. path) already contain the namespace in its value, but for those that hold a JSON, I had no idea where to keep the namespace. E.g.:

```html
<div
  data-wp-on--click="custom-ns::actions.clickHandler"
  data-wp-context='custom-ns::{ "count": 0 }'
> ... </div>
```

As we parse the JSON string inside `toVdom()`, I ended up using a `Symbol()` to store the namespace directly in the resulting object.

@luisherranz, let me know what you think.

Reply:
Oh yeah, we need to store a namespace _per_ directive. That's right.

I'm not sure I like to expose a Symbol as an API: `[ context[ $$namespace ] || namespace ]: context`. It's weird and not easy to follow.

Maybe we need to rethink the directives structure to something more like this:

```html
<div data-wp-interactive='{ "namespace": "myPlugin" }'>
  <div
    data-wp-context='{ "isOpen": true }'
    data-wp-context--context2='otherPlugin::{ "isOpen": false }'
  >
    ...
  </div>
</div>
```

```js
// what directives("context", { directives }) receives
[
  {
    namespace: "myPlugin",
    suffix: "default",
    value: {
      isOpen: true,
    },
  },
  {
    namespace: "otherPlugin",
    suffix: "context2",
    value: {
      isOpen: false,
    },
  },
];
```

Reply:
If we store namespaces inside the directives themselves, maybe there's no need to store them in the components.

Reply:
That would make sense. Let's see how complex it becomes to consume a `directives` property with that format.

Reply:
The suggested format for parsed directives was implemented in e4180a7ebf89456e1386e9253e7e2359ae8e36a8.

I also changed `evaluate()` to receive the whole directive definition instead of just the path.